rpc/core/blocks.go

```diff
@@ -333,6 +332,10 @@ func EncodeDataRootTuple(height uint64, dataRoot [32]byte) ([]byte, error) {
 	return append(paddedHeight, dataRoot[:]...), nil
 }
 
+// DataCommitmentBlocksLimit The maximum number of blocks to be used to create a data commitment.
+// It's a local parameter to protect the API from creating unnecessarily large commitments.
+const DataCommitmentBlocksLimit = 10000
```

Comment:
2048 because of the number of leaves are a power of 2. So to support 4hours batches, we need to increase to 2048.
However, to give some wiggle room and not have to change this parameter so often, we opted for 10_000

Reply:
Out of curiosity, if leaves must be a power of 2 why not use a power of 2 here (e.g. `4096` or `8192`)? 

- 2048 block * 12 seconds per block / 60 seconds per minute / 60 minutes per hours ~= 6.82 hours
- 4096 block * 12 seconds per block / 60 seconds per minute / 60 minutes per hours ~= 13.65 hours
- 8192 block * 12 seconds per block / 60 seconds per minute / 60 minutes per hours ~= 27.30 hours

If this limit is to protect the API, do we have data on the maximum number of blocks the API can reliably handle?

Reply:
> if leaves must be a power of 2 why not use a power of 2 here

It's a simple value to avoid DOSing the API with unnecessary requests. It can be a power of 2, or it can be anything, really.

> If this limit is to protect the API, do we have data on the maximum number of blocks the API can reliably handle?

Generally, the nodes wouldn't really suffer from this. But it's better to have some bounds for the API

Reply:
IMO it would still be really nice to document _why_ 10_000 was chosen in the code. Here's an example of what I'd like to see:

```suggestion
// dataCommitmentBlocksLimit is the maximum number of blocks that can be used to
// create a data commitment. This is a local parameter to protect the API from
// creating unnecessarily large commitments.
const dataCommitmentBlocksLimit = 10_000 // ~33 hours of blocks assuming 12 second blocks.
```

if it's a local param, should it be unexported?